presto-thrift-interface/src/main/java/com/facebook/presto/thrift/interfaces/client/ThriftColumnData.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.thrift.interfaces.client;
+
+import com.facebook.swift.codec.ThriftConstructor;
+import com.facebook.swift.codec.ThriftField;
+import com.facebook.swift.codec.ThriftStruct;
+
+import javax.annotation.Nullable;
+
+import static com.facebook.swift.codec.ThriftField.Requiredness.OPTIONAL;
+import static java.util.Objects.requireNonNull;
+
+@ThriftStruct
+public final class ThriftColumnData
+{
+    private final boolean[] nulls;
+    private final long[] longs;
+    private final int[] ints;
+    private final byte[] bytes;
+    private final double[] doubles;
+    private final String columnName;
+
+    @ThriftConstructor
+    public ThriftColumnData(
```

Comment:
I'd like to discuss it in more details. In short, it seems to be easier/more convenient to represent a list of columns as a list of this structs. Also, it uses less thrift structs for the same purpose.